src/composeappmanager.cc

```diff
@@ -229,6 +234,7 @@ ComposeAppManager::AppsContainer ComposeAppManager::getAppsToUpdate(const Uptane
     if (!app_engine_->isFetched({app_name, app_pair.second}) || !app_engine_->isRunning({app_name, app_pair.second})) {
       // an App that is supposed to be installed and running is not fully installed or running
       apps_to_update.insert(app_pair);
+      apps_and_reasons.push_back(app_pair.first + " (not fetched / not running)");
```

Comment:
I would break this "if" into two parts to detect the exact reason, whether app is not fetched or not running. Then the log message will be more precise, e.g.

```
 - app01: not fetched
 - app02: not running
```

Reply:
Done, now we have 2 ifs